### PR TITLE
Component/form

### PR DESCRIPTION
**Pull request by iva585**
- created a form component
- added function to the component
- created story for the component

### CODE REVIEW

File: src/app/reducers/shoppingListReducer.ts [hunk 1]
```diff
@@ -24,11 +24,28 @@ const shoppingListSlice = createSlice({
     deleteShoppingListItem(state, action: PayloadAction<ShoppingListItemType>) {
       return [...state].filter((item) => item.id !== action.payload.id);
     },
+
+    addShoppingListItem(
+      state,
+      action: PayloadAction<ShoppingListItemType['title']>
+    ) {
+      const id: ShoppingListItemType['id'] = 6;
```
Comment (iva585) [inline src/app/reducers/shoppingListReducer.ts:32]: This seems to be a mistake. 
The id has to be a unique key value.

**Review by iva585 — COMMENTED** [review]
Change id to be a unique key value.

File: src/app/components/ListForm/ListForm.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import ListForm from '.';
```
Comment (F-Kirchhoff) [inline src/app/components/ListForm/ListForm.stories.tsx:1]: this import wont work

File: src/app/components/ListForm/index.tsx [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  const dispatch = useDispatch();
+
+  const [inputValue, setInputValue] = useState<string>('');
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(event.target.value);
+  };
+
+  const handleAdd = () => {
+    dispatch(addShoppingListItem(inputValue));
```
Comment (F-Kirchhoff) [inline src/app/components/ListForm/index.tsx:20]: wow, looks quite advanced

File: src/app/pages/ShoppingList/ShoppingList.tsx [hunk 4]
```diff
@@ -23,13 +24,14 @@ export default (): JSX.Element => {
           Shopping List
         </Typography>
       </AppBar>
-      <List>
+      <List sx={{ pt: 7 }}>
         {[...shoppingListItems]
           .sort((a, b) => (a.checked ? 1 : 0) - (b.checked ? 1 : 0))
```
Comment (F-Kirchhoff) [inline src/app/pages/ShoppingList/ShoppingList.tsx:29]: nice!

**Review by F-Kirchhoff — APPROVED** [review]
looks good, I have just two little suggestions.

File: src/app/components/ListForm/ListForm.stories.tsx (hunk 2 above)
Comment (iva585) [reply src/app/components/ListForm/ListForm.stories.tsx:1]: The import works, but thanks to this comment I noticed that I am missing a Provider 🥲 
Storybook with Redux can#t work without one, so thank you very much 🎉

File: src/app/components/ListForm/index.tsx (hunk 3 above)
Comment (iva585) [reply src/app/components/ListForm/index.tsx:20]: not really 😬 it is just redux, but thanks!

File: src/app/reducers/shoppingListReducer.ts [hunk 5]
```diff
@@ -24,11 +24,28 @@ const shoppingListSlice = createSlice({
     deleteShoppingListItem(state, action: PayloadAction<ShoppingListItemType>) {
       return [...state].filter((item) => item.id !== action.payload.id);
```
[the remark this answers is not in the document]
Comment (iva585) [reply src/app/reducers/shoppingListReducer.ts:25]: thanks :)

**Pull request merged**